TribalWars/src/ApplicationCore/Services/BuildingService.cs

```diff
@@ -0,0 +1,56 @@
+﻿using ApplicationCore.Entities;
+using ApplicationCore.Interfaces.Repository;
+using ApplicationCore.Interfaces.Services;
+using System.Threading.Tasks;
+
+namespace ApplicationCore.Services
+{
+    public class BuildingService : IBuildingService
+    {
+        private readonly IVillageBuildingRepository _villageBuildingRepository;
+        private readonly IBuildingRepository _buildingRepository;
+
+        public BuildingService(IVillageBuildingRepository villageBuildingRepository, IBuildingRepository buildingRepository)
+        {
+            _villageBuildingRepository = villageBuildingRepository;
+            _buildingRepository = buildingRepository;
+        }
+
+        public async Task<int> CurrentBuildingLevel(int villageId, int buildingId)
+        {
+            var building = await _villageBuildingRepository.GetVillageBuilding(villageId, buildingId);
+            return building != null ? building.CurrentLevel : 0;
```

Comment:
You can use null-coalescing operator `??` here, so the line would look like this
`return building?.CurrentLevel ?? 0;